vantage/cost_report_resource.go

```diff
@@ -0,0 +1,233 @@
+package vantage
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	modelsv2 "github.com/vantage-sh/vantage-go/vantagev2/models"
+	costsv2 "github.com/vantage-sh/vantage-go/vantagev2/vantage/costs"
+)
+
+type CostReportResource struct {
+	client *Client
+}
+
+func NewCostReportResource() resource.Resource {
+	return &CostReportResource{}
+}
+
+type CostReportResourceModel struct {
+	Token             types.String `tfsdk:"token"`
+	Title             types.String `tfsdk:"title"`
+	FolderToken       types.String `tfsdk:"folder_token"`
+	Filter            types.String `tfsdk:"filter"`
+	SavedFilterTokens types.List   `tfsdk:"saved_filter_tokens"`
+	WorkspaceToken    types.String `tfsdk:"workspace_token"`
+}
+
+func (r *CostReportResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_cost_report"
+}
+
+func (r CostReportResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"title": schema.StringAttribute{
+				MarkdownDescription: "Title of the Cost Report",
+				Required:            true,
+			},
+			"folder_token": schema.StringAttribute{
+				MarkdownDescription: "Token of the folder this report resides in.",
+				Optional:            true,
+				Computed:            true,
+			},
+			"filter": schema.StringAttribute{
+				MarkdownDescription: "Filter query to apply to the Cost Report",
+				Optional:            true,
+				Computed:            true,
+			},
+			"saved_filter_tokens": schema.ListAttribute{
+				ElementType:         types.StringType,
+				MarkdownDescription: "Saved filter tokens to be applied to the report.",
+				Optional:            true,
+				Computed:            true,
+			},
+			"workspace_token": schema.StringAttribute{
+				MarkdownDescription: "Workspace token to add the cost report to.",
+				Optional:            true,
+				Computed:            true,
+				PlanModifiers: []planmodifier.String{
```

Comment:
This modifier means that if a field is not set, use the value in the state to evaluate changes. Mainly to ensure that on create, if a user isn't specifying the workspace token and we set one (because the API returned the workspace token since it filled it in with the default workspace) then use that going forward when diff-ing the resource with API to see if there's been drift.